client/src/components/view-prompts.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+This software is Copyright ©️ 2024 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+
+import React, { useContext } from "react";
+
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  TextField,
+} from "@mui/material";
+import { Close as CloseIcon } from "@mui/icons-material";
+import CogenerationContext from "context/cogeneration";
+export interface SimpleDialogProps {
+  open: boolean;
+  selectedPrompt: string;
+  onClose: (value: string) => void;
+  setSelectedPrompt: React.Dispatch<React.SetStateAction<string>>;
+}
+
+function ViewPrompts(props: SimpleDialogProps): JSX.Element {
+  const context = useContext(CogenerationContext);
+  if (!context) {
+    throw new Error("SomeComponent must be used within a CogenerationProvider");
+  }
+  const { onClose, selectedPrompt, open, setSelectedPrompt } = props;
+
+  const handleClose = () => {
+    onClose(selectedPrompt);
+  };
+
+  const getDefaultValues = (type: string) => {
```

Comment:
should add return typing to this function to ensure that the default return values are always type asserted